functions/venue.js

```diff
@@ -702,6 +709,30 @@ exports.updateVenue_v2 = functions.https.onCall(async (data, context) => {
   admin.firestore().collection("venues").doc(venueId).update(updated);
 });
 
+exports.updateTables = functions.https.onCall((data, context) => {
+  checkAuth(context);
+
+  const venueRef = admin.firestore().collection("venues").doc(data.venueId);
+
+  return admin.firestore().runTransaction(async (transaction) => {
+    const venueDoc = await transaction.get(venueRef);
+
+    if (!venueDoc.exists) {
+      throw new HttpsError("not-found", `venue ${venueId} does not exist`);
+    }
+
+    const venueTables = [...data.tables];
+
+    const currentTableIndex = venueTables.findIndex(
+      (table) => table.reference === data.updatedTable.reference
+    );
+
+    venueTables[currentTableIndex] = data.updatedTable;
```

Comment:
The value will be set for index `-1` but it won't be read because arrays start from `0`. However, I'm not sure how firebase will react to this. I'll probably wrap it in an `if` statement and check if `currentTableIndex` is `>= 0`

Reply:
Actually I'm going to add something like this:

```tsx
    if (!currentTableIndex) {
      throw new HttpsError("not-found", `current table does not exist in the venue`);
    }
```

Reply:
> Actually I'm going to add something like this

Yup, that looks like a good solution here. Please add the commit hash before resolving comment threads.

Reply:
I got curious. This shows what would happen. Definitely worth guarding against it as you already suggested.

![B13E6D73-3A90-4E99-A0FC-C0D952514808](https://user-images.githubusercontent.com/753891/119501437-34a22c00-bdac-11eb-964b-9173bf22763c.png)

Reply:
Oh, can’t use the guard as literally stated above though, as 0 is falsy so will fail, yet is a valid index.

Reply:
Fixed in: https://github.com/sparkletown/sparkle/pull/1377/commits/eabf5f4a15571cf2e4e13c2c78ce8659071752c3

Reply:
Yeah, I have to check if it's `> 0`